### PR TITLE
Don't ban remote peer when ckb_vm received Ctrl-C signal when processing chunk_tx

### LINKED ISSUE #4701: Avoid Banning Peer on `ckb_vm::Error::External` Triggered by Ctrl-C Signal
## Bug Report

```
2024-10-29 02:02:47.915 +00:00 GlobalRt-20 INFO ckb_tx_pool::process  after_process Byte32(0x58aff34dd6bbac94032aee41ab59fa6232c4139ea135ffab7f7ed014eb89ac12) SessionId(588) remote reject: Verification failed Script(TransactionScriptError { source: Inputs[0].Type, cause: VM Internal Error: External("stopped") }) 
2024-10-29 02:02:47.915 +00:00 GlobalRt-20 INFO ckb_network::network  Ban peer "/ip4/54.168.12.92/tcp/30967/p2p/QmYx1orST2KkEbwmwGEkpSvcJjVenhKEphXhci85zcg6VX" for 259200 seconds, reason: reject Verification failed Script(TransactionScriptError { source: Inputs[0].Type, cause: VM Internal Error: External("stopped") })
```

https://github.com/nervosnetwork/ckb/blob/8cb49e4d727f337a2c80e210507a9e277bab3391/script/src/verify.rs#L1200-L1204

When chunk_run_with_signal receives ChunkCommand::Stop, it raises ckb_vm::Error::External, leading NetworkState to ban the peer. Since this error can result from a Ctrl-C signal, banning the peer is unnecessary.

### Expected Behavior
Handle ckb_vm::Error::External triggered by signals (e.g., Ctrl-C) without banning the peer.

### Environment

- **CKB version**: 0.119.0
- **Chain**: testnet
- **Operating system**: [Ubuntu 20.04, macOS 11.4, Windows 10]
- **Arch**: x64 [the output of `uname -a`]
- **Installation**: GitHub Release

### PR DESCRIPTION
**Pull request by eval-exec**
### What problem does this PR solve?

Issue Number: close #4701 

### Related changes

- let `ckb_vm::Error::External("stopped")` return `ErrorKind::Internal` instead of `ErrorKind::Script`

return `ErrorKind::Internal` then ckb will not think the tx is malformled:
https://github.com/nervosnetwork/ckb/blob/8cb49e4d727f337a2c80e210507a9e277bab3391/util/types/src/core/tx_pool.rs#L68-L84
### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code ci-runs-only: [ quick_checks,linters ]

Side effects

- None

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Comment by chenyukang** [comment]
seems not a proper PR title.

**Comment by eval-exec** [comment]
> seems not a proper PR title.

Got it, I'll update the title. Any suggestions?

**Comment by eval-exec** [comment]
Need to create a new Error to handle this case:`InternalErrorKind::Interrupts`

File: script/src/verify.rs [hunk 1]
```diff
@@ -1180,6 +1180,7 @@ where
         let mut scheduler = Scheduler::new(tx_data, version, self.syscalls_generator.clone());
         let map_vm_internal_error = |error: VMInternalError| match error {
             VMInternalError::CyclesExceeded => ScriptError::ExceededMaximumCycles(max_cycles),
+            VMInternalError::External(reason) if reason.eq("stopped") => ScriptError::Interrupts,
```
[the remark this answers is not in the document]
Comment (eval-exec) [reply script/src/verify.rs:1183]:
```
script/src/syscalls/debugger.rs
48:            .map_err(|e| VMError::External(format!("String from buffer {e:?}")))?;
script/src/syscalls/mod.rs
126:            _ => Err(Error::External(format!("CellField parse_from_u64 {i}"))),
148:            _ => Err(Error::External(format!("HeaderField parse_from_u64 {i}"))),
164:            _ => Err(Error::External(format!("InputField parse_from_u64 {i}"))),
197:            _ => Err(Error::External(format!("SourceEntry parse_from_u64 {i}"))),
243:            _ => Err(Error::External(format!("Place parse_from_u64 {i}"))),
```

**Review by chenyukang — DISMISSED** [review]

**Comment by eval-exec** [comment]
> how can we add a unit test for this, it seems hard but worth having a try.

Sure, I will add unit test.

**Review by zhangsoledad — APPROVED** [review]

**Pull request merged**